platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/DataFileCollection.java

```diff
@@ -447,6 +448,14 @@ public synchronized List<Path> compactFiles(
                             serializationVersion, reader.readDataItemBytes(fileOffset));
                     // update the index
                     index.putIfEqual(path, dataLocation, newLocation);
+                } catch (final ClosedByInterruptException e) {
+                    logger.info(
+                            EXCEPTION.getMarker(),
+                            "Failed to copy data item {} / {} due to thread interruption",
+                            fileIndex,
+                            fileOffset,
+                            e);
+                    throw e;
```

Comment:
Should not be rethrown.

Reply:
That was my initial thinking but then I realized that we re-throw here to end `forEach` cycle.